src/intervals/queue.jl

```diff
@@ -0,0 +1,72 @@
+# Queue
+# =====
+
+type Queue{T}
```

Comment:
This is a usual queue data structure not specific to Bio.jl. This is very similar to `Bio.Intervals.StreamBuffer` but internals are slightly different (for example, `StreamBuffer` is index-oriented but `Queue` is iterator-oriented, which makes the algorithm simpler.) I've tried `Queue` of DataStructures.jl but it is not so efficient as I expected and requires a few patches to use.